### PR TITLE
movie dir as setting fixes #24

### LINKED ISSUE #24: Watchfolder as a Setting
Directories that are scanned for media should be stored as settings, not with symlinks.

### PR DESCRIPTION
**Pull request by HParker**
@XanderStrike let me know what you think of this. I haven't tested the rake task yet, just want to get what you think of going in this direction.

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -45,6 +45,9 @@ def settings
     @admin = Setting.get(:admin) || Setting.create(name: 'admin', content: '', boolean: false)
     @admin_pass = Setting.get('admin-pass') || Setting.create(name: 'admin-pass', content: '')
 
+    movie_dirs = Setting.where(name: 'movie-dir')
+    @movie_dirs = movie_dirs.count > 0 ? Setting.where(name: 'movie-dir') : [Setting.create(name: 'movie-dir', content: "#{rails.root}/public/movies/")]
```
Comment (XanderStrike) [inline app/controllers/home_controller.rb]: Rails ought to be capitalized like this: `Rails.root`

File: lib/tasks/scan_movies.rake [hunk 2]
```diff
@@ -22,9 +22,15 @@ namespace :scan do
 
     formats = ['mp4', 'avi', 'xvid', 'divx', 'mts', 'mpeg', 'mkv', 'wmv', 'ogv', 'webm', 'mov', 'mpg', 'mpe', 'm4v', 'h264', 'avchd']
 
+
+
     # get file list
-    files = `find public/movies/ -type f`.split("\n").map {|f| f.gsub('public/movies/', '')}
-    files = files.reject {|f| !formats.any? {|w| f =~ /#{w}/ }}
+    movie_dirs = Settings.where(name: 'Movie-dir')
```
Comment (XanderStrike) [inline lib/tasks/scan_movies.rake]: It's `Setting.where`. I know it's wrong I just haven't gotten to fixing it.

**Comment by XanderStrike** [comment]
I really like this, it's a good start. The Rake task still has a lot to get done, it doesn't seem to work even with setting fixed, but your approach may be sound. 

I'd say start with just one directory initially. Remember that to give the webserver access to the directory have to create a symlink to the directory in the `public`, so it may be simpler to just use that setting create the directory.

Also tests are totally a thing now, we've got continuous integration and everything now.

**Comment by HParker** [comment]
@XanderStrike I have totally forgotten the state of this. I am guessing we can close it and delete the branch. Do you remember?

**Comment by XanderStrike** [comment]
Yeah we did this on another branch.

**Pull request closed without merging**